### PR TITLE
Integrating refiner containerization into TTC workflows

### LINKED ISSUE #270: Add github workflow to publish lambda images to GHCR
### Description
APHL has asked us to make dummy instances of the TTC and augmentation lambdas available to them by 3/3 so that they can test out integrating the lambdas into their workflows. We need to update our build + publish [workflow](https://github.com/CDCgov/dibbs-text-to-code/blob/main/.github/workflows/docker-build-ghcr.yml) to include both TTC and augmentation lambdas. 

### Notes
It may be worth looking at what Refiner has done for their workflow, [here](https://github.com/CDCgov/dibbs-ecr-refiner/blob/main/.github/workflows/docker-image-push.yml), to see if we can reuse it. Jake would also be a good resource if you have additional questions.

For now, APHL just needs dummy versions of the lambdas for testing purposes so feel free to add them to the repo.

### PR DESCRIPTION
**Pull request by robertandremitchell**
## Description

Adds the lambda and augmentation lambdas, initial make of a Dockerfile.

I'm not entirely sure the best way to test, but work based off ecr-refiner work here:

- https://github.com/CDCgov/dibbs-ecr-refiner/blob/main/.github/workflows/docker-image-push.yml
- https://github.com/CDCgov/dibbs-ecr-refiner/blob/main/.github/workflows/build.yml
## Related Issues

Closes #270

## Additional Notes

[Add any additional context or notes that reviewers should know about.]

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist

Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers

Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: Dockerfile.augmentation [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# Install build tools for compiling C++ extensions
+RUN microdnf install -y gcc-c++ make && microdnf clean all
+
+# Copy over just the pyproject.toml file and install the dependencies doing this
+# before copying the rest of the code allows for caching of the dependencies
+COPY ./pyproject.toml ${LAMBDA_TASK_ROOT}/pyproject.toml
+RUN pip install --no-cache-dir "$(printf '%s' ".[${ENVIRONMENT}]")"
+# Improve Lambda startup time by validating SpaCy models upfront,
+# which preloads necessary libraries during build
+RUN python -m spacy validate
```
Comment (m-goggins) [inline Dockerfile.augmentation]: This can be removed since we don't use spacy for augmentation

File: Dockerfile.augmentation [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# Copy over the rest of the code
+COPY ./src ${LAMBDA_TASK_ROOT}/src/augmentation_lambda
+COPY README.md ${LAMBDA_TASK_ROOT}/README.md
+
+ENV PYTHONPATH="${LAMBDA_TASK_ROOT}/src/augmentation_lambda"
+
+CMD [ "augmentation_lambda.main.handler" ]
```
Comment (m-goggins) [inline Dockerfile.augmentation]:
```suggestion
CMD [ "augmentation_lambda.lambda_function.handler" ]
```

Comment (m-goggins) [reply Dockerfile.augmentation]: ^I know our TTC Dockerfile has it as main but my PR updates it to `lambda_function`, which is the more standard convention.

File: .github/workflows/docker-build-ghcr.yml [hunk 3]
```diff
@@ -19,16 +19,65 @@ permissions:
...
+      - lambda-build
     permissions:
       contents: read
       packages: write
       id-token: write
 
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v6
+        uses: actions/checkout@v4
```
[the remark this answers is not in the document]
Comment (robertandremitchell) [reply .github/workflows/docker-build-ghcr.yml]: no, this was just a symptom of me missing it when integrating the new code from refiner. updated!

File: .github/workflows/docker-image-push.yml [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+              # [ "$PUSH_TTC" = "true" ] && TAGS+=("$TTC_REGISTRY/dibbs-text-to-code/${BASES[$IMAGE]}:latest")
+            fi
+
+            # Output as comma-separated string
+            echo "${IMAGE,,}_tags=$(IFS=, ; echo "${TAGS[*]}")" >> $GITHUB_OUTPUT
+          done
+
+      - name: Debug Docker tags
+        run: |
+          echo "APP_TAGS: ${{ steps.vars.outputs.app_tags }}"
```
[the remark this answers is not in the document]
Comment (robertandremitchell) [reply .github/workflows/docker-image-push.yml]: removed!
